File: other/comparing_floats_2012/01_calculate_one.c
#include <stdio.h>

int main() {
    float f = 0.1f;
    float sum;
    sum = 0;

    for (int i = 0; i < 10; ++i)
        sum += f;
    float product = f * 10;
    printf("sum  = %1.15f\nmul1 = %1.15f\nmul2 = %1.15f\n",
            sum, product, f * 10);
}

File: other/comparing_floats_2012/01_calculate_one.nim
import strformat

const f: float32 = 0.1
var sum: float32
sum = 0

for i in 1..10:
  sum += f
let product: float32 = f * 10
echo &"sum  = {sum:1.15f}\nmul1 = {product:1.15f}\nmul2 = {(block:f * 10):1.15f}"
